dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -639,18 +653,54 @@ public Boolean checkCWL(File content) {
         //go through each line of the file content and find the word patterns as described above
         try {
             List<String> fileContent = java.nio.file.Files.readAllLines(p, StandardCharsets.UTF_8);
-            for (String line : fileContent) {
+            for (int i = 0; i < fileContent.size(); i++) {
+                String line = fileContent.get(i);
+
                 Matcher matchWf = classWfPattern.matcher(line);
                 Matcher matchTool = classToolPattern.matcher(line);
                 Matcher matchInput = inputPattern.matcher(line);
-                Matcher matchOutput = outputPattern.matcher(line);
+
+                Matcher matchOutputEmpty = outputPatternEmpty.matcher(line);
+                Matcher matchOutputParas = outputPatternParas.matcher(line);
+
                 Matcher matchCommand = commandPattern.matcher(line);
                 Matcher matchVersion = versionPattern.matcher(line);
                 Matcher matchSteps = stepsPattern.matcher(line);
+
                 if (matchInput.find() && !stepsFound) {
                     inputFound = true;
-                } else if (matchOutput.find()) {
+                } else if (matchOutputEmpty.find()) {
                     outputFound = true;
+                } else if (matchOutputParas.find()) {
+                    outputFound = true;
+
+                    i++;
+
+                    try {
+                        line = fileContent.get(i);
+                    } catch (IndexOutOfBoundsException e) {
+                        missing += " valid list for 'outputs' at line " + --i;
```

Comment:
Avoid using `--1` and `i++` above. It can be quite confusing to use these in the middle of a for-loop.